app/models/state_file_archived_intake_request.rb

```diff
@@ -15,19 +17,80 @@
 #
 #  idx_on_state_file_archived_intake_id_7dd0f99380  (state_file_archived_intake_id)
 #
+require 'csv'
+
 class StateFileArchivedIntakeRequest < ApplicationRecord
   devise :lockable, unlock_in: 60.minutes, unlock_strategy: :time
   has_many :state_file_archived_intake_access_logs, class_name: 'StateFileArchivedIntakeAccessLog'
   belongs_to :state_file_archived_intake, class_name: 'StateFileArchivedIntake', optional: true
 
+  before_create :populate_fake_addresses
+
   def self.maximum_attempts
     2
   end
 
   def increment_failed_attempts
     super
-    if attempts_exceeded? && !access_locked?
-      lock_access!
+    lock_access! if attempts_exceeded? && !access_locked?
+  end
+
+  def fake_addresses
```

Comment:
why did you choose to add the logic for fetching random addresses on the request class itself? as opposed to creating a separate class/service for this?

Reply:
because this is only used here, If it was used somewhere else  I would have made a service.

Reply:
i don't think this needs to be handled in this PR, but...even if it's not used elsewhere, i think a more object oriented approach would be to have this in it's own class. we could also then test some of meatier logic in these methods individually. 

...something to keep in mind for the future!